src/devices/Bh1745/Bh1745.cs

```diff
@@ -0,0 +1,430 @@
+                intSource = (byte)((intSource & (byte)~Mask.INT_SOURCE) | (byte)value << 2);
+
+                Write8BitsToRegister((byte)Register.INTERRUPT, intSource);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets whether the interrupt pin is enabled.
+        /// </summary>
+        public bool InterruptIsEnabled
+        {
+            get
+            {
+                var intPin = Read8BitsFromRegister((byte)Register.INTERRUPT);
+                intPin = (byte)(intPin & (byte)Mask.INT_ENABLE);
+                return Convert.ToBoolean(intPin);
+            }
+            set
+            {
+                var intPin = Read8BitsFromRegister((byte)Register.INTERRUPT);
+                intPin = (byte)((intPin & (byte)~Mask.INT_ENABLE) | Convert.ToByte(value));
+
+
+                Write8BitsToRegister((byte)Register.INTERRUPT, intPin);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the persistence function of the interrupt.
+        /// </summary>
+        public InterruptPersistence InterruptPersistence
+        {
+            get
+            {
+                var intPersistence = Read8BitsFromRegister((byte)Register.PERSISTENCE);
+                intPersistence = (byte)(intPersistence & (byte)Mask.PERSISTENCE);
+                return (InterruptPersistence)intPersistence;
+            }
+            set
+            {
+                var intPersistence = Read8BitsFromRegister((byte)Register.PERSISTENCE);
+                intPersistence = (byte)((intPersistence & (byte)~Mask.PERSISTENCE) | (byte)value);
+
+                Write8BitsToRegister((byte)Register.PERSISTENCE, intPersistence);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the lower interrupt threshold.
+        /// </summary>
+        public ushort LowerInterruptThreshold
+        {
+            get => Read16BitsFromRegister((byte)Register.TL);
+            set => WriteShortToRegister((byte)Register.TL, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the higher interrupt threshold.
+        /// </summary>
+        public ushort HigherInterruptThreshold
+        {
+            get => Read16BitsFromRegister((byte)Register.TH);
+            set => WriteShortToRegister((byte)Register.TH, value);
+        }
+
+        /// <summary>
+        /// Gets whether the last measurement is valid.
+        /// </summary>
+        public bool MeasurementIsValid
+        {
+            get
+            {
+                var valid = Read8BitsFromRegister((byte)Register.MODE_CONTROL2);
+                valid = (byte)(valid & (byte)Mask.VALID);
+                return Convert.ToBoolean(valid);
+            }
+        }
+
+        /// <summary>
+        /// Gets the data in the red data register.
+        /// </summary>
+        public ushort RedDataRegister => Read16BitsFromRegister((byte)Register.RED_DATA);
```

Comment:
I think all of these should instead be methods instead of readonly properties. See #626 for more info